fresnel/geometry.py

```diff
@@ -183,6 +183,157 @@ def radius(self):
     def color(self):
         return util.array(self._geometry.getColorBuffer(), geom=self)
 
+
+class Box(Cylinder):
+    R""" Box geometry.
+
+    Generate a "box" shape.
+
+    Args:
+        scene (:py:class:`fresnel.Scene`): Add the geometry to this scene
+        box (`numpy.ndarray` or `array_like`): (``1x1, 1x3, or 1x6`` : ``float32``) Assumes 1x1 is cubic, 1x3 is orthorhombic, and 1x6 is triclinic.
+        radius (`numpy.ndarray` or `array_like`): (``N`` : ``float32``) - Radius of box edges.
+        color (`numpy.ndarray` or `array_like`): (``Nx2x3`` : ``float32``) - Color of box edges.
+
+    .. seealso::
+
+        :doc:`examples/01-Primitives/05-Box-geometry`
+            Tutorial: defining and setting box geometry properties
+        :doc:`examples/02-Advanced-topics/05-GSD-visualization`
+            Tutorial: Visualizing GSD files
+
+
+    Note:
+        The constructor arguments ``radius``, and ``color`` are optional. If you do not provide them,
+        they are initialized to 0.5 and black, respectively.
+
+    .. hint::
+        Avoid costly memory allocations and type conversions by specifying primitive properties in the appropriate
+        numpy array type.
+
+    Attributes:
+        points (:py:class:`fresnel.util.array`): Read or modify the box.
+        radius (:py:class:`fresnel.util.array`): Read or modify the radii of the box edges.
+        color (:py:class:`fresnel.util.array`): Read or modify the colors of the box.
+    """
+
+    def __init__(self,
+                 scene,
+                 box,
+                 radius=None,
+                 color=None,
+                 material=material.Material(solid=1.0, color=(0, 0, 0)),
+                 outline_material=material.Material(solid=1.0, color=(0, 0, 0)),
+                 outline_width=0.0):
+
+        self._geometry = scene.device.module.GeometryCylinder(scene._scene, 12)
```

Comment:
Why not `super().__init__(scene=scene, N=12)`?. Then you inherit methods and properties from Cylinder and can add the `box` property and define the convenient initialization routines.

Reply:
done!